src/ts/components/listselector.ts

```diff
@@ -10,11 +10,45 @@ export interface ListItem {
   label: string;
 }
 
+/**
+ * Filter function that can be used to filter out list items added through {@link ListSelector.addItem}.
+ *
+ * Using this does not make a lot of sense when items are added from outside because the added items can be
+ * controlled anyway, but it does make sense to be used in conjunction with subclasses that populate
+ * themselves automatically via the player API, e.g. {@link SubtitleSelectBox}.
+ */
+export interface ListItemFilter {
+  /**
+   * Takes a list item and decides whether it should pass or be discarded.
+   * @param {ListItem} listItem the item to apply the filter to
+   * @returns {boolean} true to let the item pass through the filter, false to discard the item
+   */
+  (listItem: ListItem): boolean;
+}
+
+/**
+ * Translator function to translate labels of list items added through {@link ListSelector.addItem}.
+ *
+ * Using this does not make a lot of sense when items are added from outside because the added items can be
+ * controlled anyway, but it does make sense to be used in conjunction with subclasses that populate
+ * themselves automatically via the player API, e.g. {@link SubtitleSelectBox}.
+ */
+export interface ListItemLabelTranslator {
+  /**
+   * Takes a list item, optionally changes the label, and returns the new label.
+   * @param {ListItem} listItem the item to translate
+   * @returns {string} the translated or original label
+   */
+  (listItem: ListItem): string;
+}
+
 /**
  * Configuration interface for a {@link ListSelector}.
  */
 export interface ListSelectorConfig extends ComponentConfig {
   items?: ListItem[];
+  filter?: ListItemFilter;
```

Comment:
Maybe do a more generic thing such as: 
```
filter?: Predicate<ListItem>;
translator: Producer<ListItem, string>;
```

Reply:
As explained above, this is the only place where these interfaces are used so I think it's not too bad if the names match.